client/hotrod-client/src/main/java/org/infinispan/client/hotrod/RemoteCache.java

```diff
@@ -352,47 +347,45 @@
    CloseableIteratorSet<Entry<K, V>> entrySet(IntSet segments);
 
    /**
-    * Synthetic operation. The client iterates over the set of keys and calls put for each one of them. This results in
-    * operation not being atomic (if a failure happens after few puts it is not rolled back) and costly (for each key in
-    * the parameter map a remote call is performed).
+    * Updates all entries in the remote cache. This operation provides better performance than calling put for each entry.
     */
    @Override
    void putAll(Map<? extends K, ? extends V> map, long lifespan, TimeUnit unit);
 
    /**
-    * Synthetic operation.
+    * Updates all entries in the remote cache.
```

Comment:
I wonder if we should word this differently, since it isn't always an update and it is only the entries provided. And why does only the prior method mention the performance?

Reply:
@wburns thanks, I've made some changes. hopefully getting closer to the mark now.